packages/contracts/test/core/dao/dao.ts

```diff
@@ -1190,98 +1219,234 @@ describe('DAO', function () {
+      await dao.grant(
+        dao.address,
+        caller.address,
+        PERMISSION_IDS.VALIDATE_SIGNATURE_PERMISSION_ID
+      );
+
+      // The caller can validate signatures now.
+      expect(await dao.connect(caller).isValidSignature(hash, signature)).to.not
+        .be.reverted;
+
+      // Because the caller is allowed unconditionally, the signature is always valid.
       expect(
-        await dao.isValidSignature(ethers.utils.keccak256('0x00'), '0x00')
-      ).to.be.eq('0x00000000');
+        await dao.connect(caller).isValidSignature(hash, signature)
+      ).to.equal(VALID_ERC1271_SIGNATURE);
+
+      // Because the other caller is not allowed, the signature is always invalid.
+      expect(
+        await dao.connect(otherCaller).isValidSignature(hash, signature)
+      ).to.equal(INVALID_ERC1271_SIGNATURE);
     });
 
-    it('should allow only `SET_SIGNATURE_VALIDATOR_PERMISSION_ID` to set validator', async () => {
-      await expect(
-        dao
-          .connect(signers[2])
-          .setSignatureValidator(ethers.Wallet.createRandom().address)
-      )
+    it('allows caller-specific signature validation conditions', async () => {
+      // Try to call with caller but caller has no permission
+      expect(await dao.connect(caller).isValidSignature(hash, signature))
         .to.be.revertedWithCustomError(dao, 'Unauthorized')
-        .withArgs(
-          dao.address,
-          signers[2].address,
-          PERMISSION_IDS.SET_SIGNATURE_VALIDATOR_PERMISSION_ID
-        );
-    });
+        .withArgs();
 
-    it('should set validator and emits event', async () => {
-      const validatorAddress = ethers.Wallet.createRandom().address;
-      const tx = await dao.setSignatureValidator(validatorAddress);
+      // Deploy a mock condition
+      const mockCondition = await mockConditionFactory.deploy();
 
-      expect(await dao.signatureValidator()).to.be.eq(validatorAddress);
+      // Grant the permission to validate signatures to the caller
+      await dao.grantWithCondition(
+        dao.address,
+        caller.address,
+        PERMISSION_IDS.VALIDATE_SIGNATURE_PERMISSION_ID,
+        mockCondition.address
+      );
 
-      await expect(tx)
-        .to.emit(dao, EVENTS.SignatureValidatorSet)
-        .withArgs(validatorAddress);
-    });
+      // The caller can validate signatures now.
+      expect(await dao.connect(caller).isValidSignature(hash, signature)).to.not
+        .be.reverted;
+
+      // Check that the mock condition will answer true.
+      expect(await mockCondition.answer()).to.be.true;
 
-    it('should call the signature validator', async () => {
-      const ERC1271MockFactory = await smock.mock('ERC1271Mock');
-      const erc1271Mock = await ERC1271MockFactory.deploy();
+      // Check that the signature is invalid in this case.
+      expect(
+        await dao.connect(caller).isValidSignature(hash, signature)
+      ).to.equal(VALID_ERC1271_SIGNATURE);
```

Comment:
`// Check that the signature is invalid in this case.` you mean `valid` right?

Reply:
Yes, fixed in b89cb57af753ef490ebe99891018035025a85d33